laborator/content/operatii-memorie-gdb/solution/3-pixels/pixel.h

```diff
@@ -0,0 +1,94 @@
+#ifndef __PIXEL_H__
+#define __PIXEL_H__
+
+typedef struct Pixel {
+	unsigned char R;
+	unsigned char G;
+	unsigned char B;
+} Pixel;
+
+typedef struct Picture {
+	int height;
+	int width;
+	Pixel **pix_array;
+} Picture;
+
+Picture *generatePicture(int height, int width, Pixel **pix_array)
+{
+	Picture *pic = malloc(sizeof(*pic));
+
+	pic->height = height;
+	pic->width = width;
+
+	pic->pix_array = malloc(pic->height * sizeof(*(pic->pix_array)));
+
+	for (int i = 0; i < pic->height; ++i)
+		pic->pix_array[i] = malloc(pic->width * sizeof(*(*(pic->pix_array))));
+
+	for (int i = 0; i < pic->height; ++i)
+		for (int j = 0; j < pic->width; ++j)
+			pic->pix_array[i][j] = pix_array[i][j];
+
+	return pic;
+}
+
+Pixel generatePixel (const unsigned char R,
+					const unsigned char G, const unsigned char B)
+{
+	Pixel pixel;
+
+	pixel.R = R;
+	pixel.G = G;
+	pixel.B = B;
+
+	return pixel;
+}
+
+void printPicture(Picture *pic)
+{
+	for (int i = 0; i < pic->height; ++i) {
+		for (int j = 0; j < pic->width; ++j) {
+			printf("%u %u %u ", pic->pix_array[i][j].R,
+						pic->pix_array[i][j].G, pic->pix_array[i][j].B);
+			if (j != pic->width - 1)
+				printf(" | ");
+		}
+		printf("\n");
+	}
+}
+
+Pixel **generatePixelArray(int height, int width)
+{
+	Pixel **pix_array = malloc(height * sizeof(*pix_array));
```

Comment:
Do some error handling in case `malloc()` returns `NULL`. You can simply `perror("malloc")` and then call `exit(1)`.

Reply:
Ditto. Not resolved.